### PR TITLE
errors replacements are labeled as faster 

### PR DESCRIPTION
**Pull request by mmorel-35**

### CODE REVIEW

**Review by ccoVeille — APPROVED** [review]

**Review by alexandear — APPROVED** [review]

File: analyzer/analyzer.go [hunk 1]
```diff
@@ -163,18 +163,16 @@ func (n *perfSprint) run(pass *analysis.Pass) (interface{}, error) {
 			removedFmtUsages[fname]++
 			if fn == "fmt.Errorf" {
 				neededPackages[fname]["errors"] = true
-				d = newAnalysisDiagnostic(
+				d = newFunctionReplacementDiagnostic(
```
Comment (catenacyber) [inline analyzer/analyzer.go:166]: Oh, I think everything should be newFasterFunctionReplacementDiagnostic

**Comment by catenacyber** [comment]
Needs a rebase now

**Review by ccoVeille — APPROVED** [review]

File: analyzer/testdata/src/p/p.go [hunk 2]
```diff
@@ -22,22 +22,22 @@ func positive() {
...
-	fmt.Sprint(err)                // want "fmt.Sprint can be replaced with err.Error()"
+	fmt.Sprintf("%s", errSentinel) // want "fmt.Sprintf can be replaced with faster errSentinel.Error()"
+	fmt.Sprintf("%v", errSentinel) // want "fmt.Sprintf can be replaced with faster errSentinel.Error()"
+	fmt.Sprint(errSentinel)        // want "fmt.Sprint can be replaced with faster errSentinel.Error()"
+	fmt.Sprintf("%s", io.EOF)      // want "fmt.Sprintf can be replaced with faster io.EOF.Error()"
+	fmt.Sprintf("%v", io.EOF)      // want "fmt.Sprintf can be replaced with faster io.EOF.Error()"
+	fmt.Sprint(io.EOF)             // want "fmt.Sprint can be replaced with faster io.EOF.Error()"
+	fmt.Sprintf("%s", err)         // want "fmt.Sprintf can be replaced with faster err.Error()"
+	fmt.Sprintf("%v", err)         // want "fmt.Sprintf can be replaced with faster err.Error()"
+	fmt.Sprint(err)                // want "fmt.Sprint can be replaced with faster err.Error()"
```
Comment (alexandear) [inline analyzer/testdata/src/p/p.go:40]: Changing the report message constitutes a breaking change. However, the PR's title states "chore: define newFasterFunctionReplacementDiagnostic," which suggests it's a simple refactoring. I propose we either change the PR's title or avoid making breaking changes.

**Comment by ccoVeille** [comment]
Why was it closed?

**Pull request closed without merging**